include/LArNDGeomSimple.h

```diff
@@ -0,0 +1,214 @@
+     *  @param  tpcID tpc id as int
+     */
+    void AddTPC(const double min_x, const double max_x, const double min_y, const double max_y, const double min_z, const double max_z, const int tpcID);
+
+    /**
+     *  @brief  Get the box surrounding all TPCs
+     *
+     *  @param  min_x minimum x position of the TPCs
+     *  @param  max_x maximum x position of the TPCs
+     *  @param  min_y minimum y position of the TPCs
+     *  @param  max_y maximum y position of the TPCs
+     *  @param  min_z minimum z position of the TPCs
+     *  @param  max_z maximum z position of the TPCs
+     */
+    void GetSurroundingBox(double &min_x, double &max_x, double &min_y, double &max_y, double &min_z, double &max_z) const;
+
+    std::map<unsigned int, LArNDTPCSimple> m_TPCs; ///< map of the TPCs keyed on their unique id
+
+private:
+};
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+LArNDGeomSimple::LArNDGeomSimple()
+{
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+inline int LArNDGeomSimple::GetTPCNumber(const pandora::CartesianVector &position) const
+{
+    for (auto const &tpc : m_TPCs)
+    {
+        if (tpc.second.IsInTPC(position))
+            return tpc.first;
+    }
+    return -1;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+inline int LArNDGeomSimple::GetModuleNumber(const pandora::CartesianVector &position) const
+{
+    for (auto const &tpc : m_TPCs)
+    {
+        if (tpc.second.IsInTPC(position))
+            return tpc.first / 2;
+    }
+    return -1;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+inline void LArNDGeomSimple::AddTPC(
+    const double min_x, const double max_x, const double min_y, const double max_y, const double min_z, const double max_z, const int tpcID)
+{
+    if (m_TPCs.count(tpcID))
+        std::cout << "LArNDGeomSimple: trying to add another TPC with tpc id " << tpcID << "! Doing nothing. " << std::endl;
+    else
+        m_TPCs[tpcID] = LArNDTPCSimple(min_x, max_x, min_y, max_y, min_z, max_z, tpcID);
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+inline void LArNDGeomSimple::GetSurroundingBox(double &min_x, double &max_x, double &min_y, double &max_y, double &min_z, double &max_z) const
+{
+    min_x = std::numeric_limits<double>::max();
+    min_y = std::numeric_limits<double>::max();
+    min_z = std::numeric_limits<double>::max();
+    max_x = std::numeric_limits<double>::min();
+    max_y = std::numeric_limits<double>::min();
+    max_z = std::numeric_limits<double>::min();
```

Comment:
I think you want either -max() here or lowest(). min returns the minimum positive floating point value rather than the lowest possible value.

Reply:
Thanks, went for -max()